src/cipher.jl

```diff
@@ -110,39 +131,146 @@ function Decryptor(name::AbstractString, key)
     if nettle_major_version >= 3
         ccall( cipher_type.set_decrypt_key, Void, (Ptr{Void}, Ptr{UInt8}), state, pointer(key))
     else
-        ccall( cipher_type.set_decrypt_key, Void, (Ptr{Void}, Cuint, Ptr{UInt8}), state, length(key), pointer(key))
+        ccall( cipher_type.set_decrypt_key, Void, (Ptr{Void}, Cuint, Ptr{UInt8}), state, sizeof(key), pointer(key))
     end
 
     return Decryptor(cipher_type, state)
 end
 
+function decrypt!(state::Decryptor, e::Symbol, iv::Array{UInt8,1}, result, data)
+    if sizeof(result) < sizeof(data)
+        throw(ArgumentError("Output array of length $(sizeof(result)) insufficient for input data length ($(sizeof(data)))"))
+    end
+    if sizeof(result) % state.cipher_type.block_size > 0
+        throw(ArgumentError("Output array of length $(sizeof(result)) must be N times $(state.cipher_type.block_size) bytes long"))
+    end
+    if sizeof(data) % state.cipher_type.block_size > 0
+        throw(ArgumentError("Input array of length $(sizeof(data)) must be N times $(state.cipher_type.block_size) bytes long"))
+    end
+    if sizeof(iv) != state.cipher_type.block_size
+        throw(ArgumentError("Iv must be $(state.cipher_type.block_size) bytes long"))
+    end
+    if ! (symbol(uppercase(string(e))) in _cipher_suites)
+        throw(ArgumentError("now supports $(_cipher_suites) only but ':$(e)'"))
+    end
+if VERSION >= v"0.4.0"
```

Comment:
In the future, ccall needs a symbol made from constant value when select nettle functions (nettle_cbc_encrypt, nettle_ctr_crypt, nettle_gcm_encrypt, etc...) to implement other block cipher modes of operation (:CBC, :CTR, :GCM, etc...).
So I want to use dlsym to make symbols dynamically. (It may be implemented with MACROs). And these nettle functions will be called by ccall much more times repeated frequently in the 'update' loop of block cipher modes. ('update' functions has not be implemented yet.) For this reason, I think it is better way to use dlsym.